### PR TITLE
Bugfix 66 add create group basic

### PR DESCRIPTION
**Pull request by RussTheAerialist**
@bryantbhowell - I think you are looking for some ability to create groups.  Groups are handled in two separate ways even though it's the same create endpoint in the server.

This handles creating basic groups, but AD group imports are a bit more complicated (because they are done in the background).  Do you need AD imports immediately?

### CODE REVIEW

File: test/assets/group_create_async.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<tsResponse xmlns="http://tableau.com/api"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://tableau.com/api http://tableau.com/api/ts-api-2.3.xsd">
+    <job id="job-id" mode="Asynchronous" type="GroupImport"
```
Comment (LGraber) [inline test/assets/group_create_async.xml]: This is there for future tests? We don't currently let people trigger an async import.

File: tableauserverclient/server/endpoint/groups_endpoint.py [hunk 2]
```diff
@@ -45,6 +45,14 @@ def delete(self, group_id):
         self.delete_request(url)
         logger.info('Deleted single group (ID: {0})'.format(group_id))
 
+    def create(self, group_item):
+        url = self.baseurl
+        create_req = RequestFactory.Group.create_req(group_item)
+        server_response = self.post_request(url, create_req)
+        new_group = GroupItem.from_response(server_response.content)[0]
```
Comment (LGraber) [inline tableauserverclient/server/endpoint/groups_endpoint.py]: Not sure what our standard is but we could probably just put "return" here.

**Comment by LGraber** [comment]
We should have a quick chat about what the api should look like for importing an AD group. I don't want to ship something too quickly and then have the signature not work for the future. Besides that ... it looks good.

File: test/assets/group_create_async.xml (hunk 1 above)
Comment (RussTheAerialist) [inline test/assets/group_create_async.xml]: Yup, that's for a future test when I implement import() (see comment above about AD import)

File: tableauserverclient/server/endpoint/groups_endpoint.py (hunk 2 above)
Comment (RussTheAerialist) [inline tableauserverclient/server/endpoint/groups_endpoint.py]: fixed

File: test/test_group.py [hunk 3]
```diff
@@ -140,3 +142,13 @@ def test_remove_user_missing_group_id(self):
         single_group._users = []
         self.assertRaises(TSC.MissingRequiredFieldError, self.server.groups.remove_user, single_group,
                           '5de011f8-5aa9-4d5b-b991-f462c8dd6bb7')
+
+    def test_create_group(self):
+        with open(CREATE_GROUP, 'rb') as f:
+            response_xml = f.read().decode('utf-8')
+        with requests_mock.mock() as m:
+            m.post(self.baseurl, text=response_xml)
+            group_to_create = TSC.GroupItem('test')
```
Comment (t8y8) [inline test/test_group.py]: Can I be high maintenance and request a non-ascii name in a test as well?

I just remember some bugs in early sample code and libs prototypes around this.

**Comment by LGraber** [comment]
:rocket:

File: test/test_group.py (hunk 3 above)
Comment (RussTheAerialist) [inline test/test_group.py]: Sure, any suggestions on appropriate words or should I just start finding interesting looking utf-8 characters?

Comment (t8y8) [inline test/test_group.py]: Got a favorite foreign language?

I used "蚵仔煎" because it's my favorite street food in the entire world...

**Comment by t8y8** [comment]
🚀 🚀

**Comment by RussTheAerialist** [comment]
Addresses creation of groups for #66 .  Importing is not currently supported.

**Pull request merged**